features/FEATURE_LWIP/lwip-interface/lwip-sys/arch/cc.h

```diff
@@ -32,6 +32,7 @@
 #ifndef __CC_H__ 
 #define __CC_H__ 
 
+#if defined(TOOLCHAIN_GCC)
```

Comment:
cc.h file is used to configure the arh.h parameters. So those typedefs will be defined by the arh.h if not defined in cc.h. Looks like only place for cc.h to needs the typedefs is memcpy function definition. So moved the needed typedef there.